python/tests/variant_tests/row_of_cmpx_type.py

```diff
@@ -0,0 +1,284 @@
+            },
+        ]
+
+
+class varnttst_read_row_of_cmpx(TstView):
+    def __init__(self):
+        # checked manually
+        self.data = [
+            {
+                "id": 0,
+                "row_arr": {
+                    "v1": ['"12, 13, 10"', '"17, 14"'],
+                    "v2": ['"hi ,"', '"how are you?"'],
+                },
+                "row_map": {
+                    "v1": {'"x"': '"bella"', '"y"': '"ciao"'},
+                    "v2": {'"a"': '"good-bye"', '"b"': '"for now!"'},
+                },
+                "row_row": {
+                    "v1": {"v11": '"the quick"', "v12": '"brown fox"'},
+                    "v2": {"v21": '"jumps over"', "v22": '"the lazy dog."'},
+                },
+                "row_udt": {
+                    "var1": {"v1": '"ferris"', "v2": '"flies"'},
+                    "var2": {"v1": '"away"', "v2": '"????"'},
+                },
+            },
+            {
+                "id": 1,
+                "row_arr": None,
+                "row_map": None,
+                "row_row": None,
+                "row_udt": None,
+            },
+            {
+                "id": 2,
+                "row_arr": {"v1": None, "v2": None},
+                "row_map": {"v1": None, "v2": None},
+                "row_row": {
+                    "v1": {"v11": None, "v12": None},
+                    "v2": {"v21": None, "v22": None},
+                },
+                "row_udt": {
+                    "var1": {"v1": None, "v2": None},
+                    "var2": {"v1": None, "v2": None},
+                },
+            },
+        ]
+        self.sql = """CREATE MATERIALIZED VIEW read_row_of_cmpx AS SELECT
+                      *
+                      FROM varnt_row_of_cmpx_tbl"""
+
+
+# Array
+class varnttst_row_of_arr_unnest_varnt_elmnt(TstView):
+    def __init__(self):
+        # checked manually
+        self.data = [
+            {"id": 0, "row1_elmnt": '"12, 13, 10"', "idx": 1},
+            {"id": 0, "row1_elmnt": '"17, 14"', "idx": 2},
+        ]
+        self.sql = """CREATE MATERIALIZED VIEW row_of_arr_unnest_varnt_elmnt AS SELECT
+                      id,  row1_elmnt, idx
+                      FROM varnt_row_of_cmpx_tbl,
+                      UNNEST(varnt_row_of_cmpx_tbl.row_arr[1]) WITH ORDINALITY AS t (row1_elmnt, idx)"""
+
+
+class varnttst_row_of_arr_field_access_varnt(TstView):
+    def __init__(self):
+        # checked manually
+        self.data = [
+            {
+                "id": 0,
+                "row_arr1": ['"12, 13, 10"', '"17, 14"'],
+                "row_arr2": ['"hi ,"', '"how are you?"'],
+            },
+            {"id": 1, "row_arr1": None, "row_arr2": None},
+            {"id": 2, "row_arr1": None, "row_arr2": None},
+        ]
+        self.sql = """CREATE MATERIALIZED VIEW row_of_arr_field_access_varnt AS SELECT
+                      id,
+                      varnt_row_of_cmpx_tbl.row_arr[1] AS row_arr1,
```

Comment:
for row values please make sure that there are tests that access them by field name as well `row_arr.v1`